app/internal/import_file.py

```diff
@@ -0,0 +1,185 @@
+        if not is_event_text_valid(event):
+            return list()
+        head, content, start_date, end_date = event.split(", ")
+        if not is_date_in_range(start_date) or \
+           not is_date_in_range(end_date.replace("\n", "")):
+            return list()
+        start_date = datetime.datetime.strptime(start_date, "%m-%d-%Y")
+        end_date = datetime.datetime.strptime(end_date.replace("\n", ""),
+                                              "%m-%d-%Y")
+        calendar_content.append({"Head": head,
+                                 "Content": content,
+                                 "S_Date": start_date,
+                                 "E_Date": end_date})
+    return calendar_content
+
+
+def import_ics_file(ics_file: str) -> List[Dict[str, Union[str, Any]]]:
+    calendar_content = []
+    with open(ics_file, "r") as ics:
+        try:
+            calendar_read = Calendar.from_ical(ics.read())
+        except (IndexError, ValueError):
+            return list()
+        for component in calendar_read.walk():
+            if component.name == "VEVENT":
+                if str(component.get('summary')) is None or \
+                   component.get('dtstart') is None or \
+                   component.get('dtend') is None or \
+                   not is_date_in_range(component.get('dtstart').dt) or \
+                   not is_date_in_range(component.get('dtend').dt):
+                    return list()
+                else:
+                    calendar_content.append({
+                        "Head": str(component.get('summary')),
+                        "Content": str(component.get('description')),
+                        "S_Date": component.get('dtstart').dt
+                        .replace(tzinfo=None),
+                        "E_Date": component.get('dtend').dt
+                        .replace(tzinfo=None)
+                    })
+    return calendar_content
+
+
+def save_events_to_database(events: List[Dict[str, Union[str, Any]]],
+                            user_id: int,
+                            session: SessionLocal = SessionLocal()) -> None:
+    """insert the events into Event table"""
+    for event in events:
+        event = Event(
+            title=event["Head"],
+            content=event["Content"],
+            start=event["S_Date"],
+            end=event["E_Date"],
+            owner_id=user_id
+        )
+        session.add(event)
+    session.commit()
+    session.close()
+
+
+def user_click_import(file: str, user_id: int,
+                      session: SessionLocal = SessionLocal()) -> str:
+    """
+    when user choose a file and click import, we are checking the file
+    and if everything is ok we will insert the data to DB
+    """
+    if is_file_valid_to_import(file):
+        if is_file_extension_valid(file, ".ics"):
+            import_file = import_ics_file(file)
+        else:
+            import_file = import_txt_file(file)
+        if import_file:
```

Comment:
You can use `and` to squash the two `if`s together and get your code to be a bit more sparse.

Reply:
Done